### PR TITLE
DMEREX-584, 585, 721 - RAD Lab Form and Using Valuesets

### PR DESCRIPTION
**Pull request by yuliumitre**
This PR creates and populates the Lab form for RAD. It also replaces hard coded codes with defined value set. I also clean up the RAD order form as well to use the valueset and sub-questionnaire. 

When test this PR, please use the corresponding test-ehr branch radlab (https://github.com/HL7-DaVinci/test-ehr/pull/35)

The test patient to use is William Oster with id pat015. 
The test results pulled from the EHR are the latest test results with quantity and unit. 

Here are some notes for sub-sections not pre-populated:
1. Provider (physician/NPP) who is performing the face-to-face evaluation - only the face-to-face encounter date is pre-populated as we can't pull the provider's detailed information 
2. Person or entity performing testing - the laboratory and tester names can only be pre-populated if the Observation.performer contains "Organization" and "Practitioner" with display
3. Signature section - since we are not certain who will sign the form as it could be providers or the testers, we decided to leave as prompt instead of pre-populating.

### CODE REVIEW

**Comment by yunwwang** [comment]
Adding the unit to the value is not good since it breaks the specific type of the value. See the screen shot: 
image
I would suggest to add the unit as part of the table header. For example, instead of "PH", change the item text to PH [pH]. Some table header already have unit in it.

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesLab.json [hunk 1]
```diff
@@ -0,0 +1,789 @@
...
+        }
+      ]
+    },
+    {
+      "linkId": "4",
+      "text": "Test Results",
+      "type": "group",
+      "item": [
+        {
+          "linkId": "5.1",
```
Comment (yunwwang) [inline RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesLab.json]: Considering align the linkId either with the parent's linkId

File: RespiratoryAssistDevices/R4/files/RespiratoryAssistDevicesPrepopulation-0.1.0.cql [hunk 2]
```diff
@@ -145,6 +55,102 @@ define RadDeviceRequested:
...
+  else
+    null  
+define "TestLaboratory":
+  if exists("TestPerformers") then 
+    "TestPerformers" P
+    where P.type = 'Organization'
+    return P.display.value
+  else
+    null  
+
```
Comment (yunwwang) [inline RespiratoryAssistDevices/R4/files/RespiratoryAssistDevicesPrepopulation-0.1.0.cql:86]: Lab NPI can be retrieved as Organization.identifier where identifier.system = 'http://hl7.org/fhir/sid/us-npi'

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesLab.json [hunk 3]
```diff
@@ -0,0 +1,789 @@
...
+              "url": "http://hl7.org/fhir/StructureDefinition/cqf-expression",
+              "valueExpression": {
+                "language": "text/cql",
+                "expression": "\"RespiratoryAssistDevicesPrepopulation\".TestLaboratory"
+              }
+            }
+          ]
+        },
+        {
+          "linkId": "3.2",
```
Comment (yunwwang) [inline RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesLab.json:114]: Lab's NPI can also be pre populated from Organization resource.

File: RespiratoryAssistDevices/R4/files/RespiratoryAssistDevicesPrepopulation-0.1.0.cql (hunk 2 above)
Comment (yuliumitre) [reply RespiratoryAssistDevices/R4/files/RespiratoryAssistDevicesPrepopulation-0.1.0.cql:86]: DTR can't get the FHIR object unless the reference is related to the patient and is directly referred from the DeviceRequest or ServiceRequest for Practitioner.

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesLab.json (hunk 3 above)
Comment (yuliumitre) [reply RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesLab.json:114]: The same as above as we can't retrieve the observation's organization as it is not related to the patient.

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevices.json [hunk 4]
```diff
@@ -270,71 +78,47 @@
           "type": "open-choice",
           "text": "Patient diagnoses for order",
           "required": false,
+          "repeats": true,
           "extension": [
             {
```
Comment (yunwwang) [inline RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevices.json:81]: This extension is not used any more.

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevices.json [hunk 5]
```diff
@@ -270,71 +78,47 @@
...
+          ],
+          "answerValueSet": "http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.122"
         },
         {
           "linkId": "4.2",
-          "text": "If Other, describe",
-          "type": "string",
-          "required": true,
-          "enableWhen": [
+          "text": "Other Diagnoses:",
```
Comment (yunwwang) [inline RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevices.json:92]: Will this be changed based on Tuesday's discussion?

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevices.json [hunk 6]
```diff
@@ -483,125 +235,22 @@
...
-              ]
-            },
-            {
-              "linkId": "5.7.3",
-              "text": "Other",
-              "type": "text"
-            }
-          ]
+          "repeats": true,
+          "answerValueSet": "http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.129"
```
Comment (yunwwang) [inline RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevices.json]: why do we change the table to a drop list?

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesLab.json [hunk 7]
```diff
@@ -0,0 +1,789 @@
...
+              {
+                "system": "http://hl7.org/fhir/questionnaire-item-control",
+                "code": "gtable"
+              }
+            ]
+          }
+        }
+      ],
+      "item": [
+        {
```
Comment (yunwwang) [inline RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesLab.json:57]: put subqestionnaire for item 2.1 to 2.4

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevices.json (hunk 6 above)
Comment (yuliumitre) [reply RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevices.json]: The supply list is defined as a valueset and we are going to use it as a drop down. We were also talking about merging the main request and supply, but different story is created to pre-populate it.

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesLab.json (hunk 7 above)
Comment (yuliumitre) [reply RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevicesLab.json:57]: This is the Lab form. I am pulling the encounter date from the DeviceRequest.encounter.reference to get the encounter date. We can't pull the encounter provider's names etc., so left it blank to be filled.

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevices.json (hunk 5 above)
Comment (yuliumitre) [reply RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevices.json:92]: It should be updated.

File: RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevices.json (hunk 6 above)
Comment (yunwwang) [reply RespiratoryAssistDevices/R4/resources/Questionnaire-R4-RespiratoryAssistDevices.json]: We are missing several columns of information, frequency, duration, quantity, refills. The dropdown list covers the first column only.

**Review by yunwwang — APPROVED** [review]
Looks good

**Pull request merged**